sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -693,6 +707,17 @@ class JDBCSuite extends QueryTest
     JdbcDialects.unregisterDialect(testH2Dialect)
   }
 
+  test("Map TINYINT to ByteType via JdbcDialects") {
+    JdbcDialects.registerDialect(testH2DialectTinyInt)
```

Comment:
Oh, @dongjoon-hyun, actually this test seems fine. The test dialect seems only scoped for this specific case.

Reply:
although it's a bit odd given https://github.com/apache/spark/pull/23400#discussion_r347704641, at least this test is scoped and won't affect other tests about unsigned `TINYINT` cases.

Reply:
Originally, I suspected this dialect is valid for `H2` database's type. 

Luckily, `TINYINT` of `H2` database seems to be also -128 to 127.
- https://www.h2database.com/html/datatypes.html#tinyint_type

In this case, I'm okay~

Reply:
Thank you for verifying this case @dongjoon-hyun